cpp/test/Ice/optional/Test.ice

```diff
@@ -6,279 +6,277 @@
+    sequence<VarStruct> VarStructSeq;
+    sequence<MyInterface*> MyInterfacePrxSeq;
+
+    sequence<byte> Serializable;
+
+    dictionary<int, int> IntIntDict;
+    dictionary<string, int> StringIntDict;
+    dictionary<int, MyEnum> IntEnumDict;
+    dictionary<int, FixedStruct> IntFixedStructDict;
+    dictionary<int, VarStruct> IntVarStructDict;
+    dictionary<int, MyInterface*> IntMyInterfacePrxDict;
+
+    ["cpp:type:Test::CustomMap<std::int32_t, std::string>"] dictionary<int, string> IntStringDict;
+
+    class MultiOptional
+    {
+        optional(1) byte a;
+        optional(2) bool b;
+        optional(3) short c;
+        optional(4) int d;
+        optional(5) long e;
+        optional(6) float f;
+        optional(7) double g;
+        optional(8) string h;
+        optional(9) MyEnum i;
+        optional(10) MyInterface* j;
+        optional(12) ByteSeq bs;
+        optional(13) StringSeq ss;
+        optional(14) IntIntDict iid;
+        optional(15) StringIntDict sid;
+        optional(16) FixedStruct fs;
+        optional(17) VarStruct vs;
+
+        optional(18) ShortSeq shs;
+        optional(19) MyEnumSeq es;
+        optional(20) FixedStructSeq fss;
+        optional(21) VarStructSeq vss;
+        optional(23) MyInterfacePrxSeq mips;
+
+        optional(24) IntEnumDict ied;
+        optional(25) IntFixedStructDict ifsd;
+        optional(26) IntVarStructDict ivsd;
+        optional(28) IntMyInterfacePrxDict imipd;
+
+        optional(29) BoolSeq bos;
+
+        optional(30) Serializable ser;
+        optional(31) string defaultedString = "";
+    }
+
+    class A
+    {
+        int requiredA = 0;
+        optional(1) int ma;
+        optional(50) int mb;
+        optional(500) int mc;
+    }
+
+    class B extends A
+    {
+        int requiredB = 0;
+        optional(10) int md;
+    }
+
+    class C extends B
+    {
+        string ss;
+        optional(890) string ms;
+    }
+
+    class WD
+    {
+        optional(1) int a = 5;
+        optional(2) string s = "test";
+    }
+
+    exception OptionalException
+    {
+        bool req = false;
+        optional(1) int a = 5;
+        optional(2) string b;
+    }
+
+    exception DerivedException extends OptionalException
+    {
+        string d1;
+        optional(600) string ss = "test";
+        string d2;
+    }
+
+    exception RequiredException extends OptionalException
+    {
+        string ss = "test";
+    }
+
+    class E
+    {
+        FixedStruct fse;
+    }
+
+    class F extends E
+    {
+        optional(1) FixedStruct fsf;
+    }
+
+    struct G1
+    {
+        string a;
+    }
+
+    struct G2
+    {
+        long a;
+    }
+
+    class G
+    {
+        optional(1) G1 gg1Opt;
+        G2 gg2;
+        optional(0) G2 gg2Opt;
+        G1 gg1;
+    }
+
+    interface Initial
+    {
+        void shutdown();
+
+        ["marshaled-result"] Object pingPong(Object o);
+
+        void opOptionalException(optional(1) int a, optional(2) string b)
```

Comment:
Here for example, this operation is very long, and so it makes more sense for it to be split up onto 2 lines.

This whole file is a good example to see how we handle super-long functions like this.
The original file's indentation was correct, except that it didn't add indentation for the outer-most module.